sympy/solvers/bivariate.py

```diff
@@ -199,7 +222,7 @@ def _solve_lambert(f, symbol, gens):
     nrhs, lhs = f.as_independent(symbol, as_Add=True)
     rhs = -nrhs
 
-    even_degrees = [i for i in degree_list(lhs) if i%2 == 0]
+    even_degrees = [i for i in power_list(lhs) if i%2 == 0 and i != 0]
```

Comment:
Here if I use `power_list(lhs, symbol)` then it raises an error `PolynomialError: 2**(2*p) contains an element of the set of generators.` Same error is raised in `degree_list` if an expression containing another set of generators. I understand from the error that the symbol should be free from another generators. @smichr , @Yathartha22 How can I fix this?